### PR TITLE
fix: check for latest transaction of the same company only

### LINKED ISSUE #37: Check for only deleting the latest Quotation/SalesOrder/SalesInvoice ignores the current company
This poses a problem for servers which are setup to use multiple Companies.

The current check forbids deleting a Quotation/SalesOrder/SalesInvoice if it is not the last one which is totally fine by itself.
But if there is a "newer" doc in another Company, the check forbids it too.

This must be changed to only look into the Company of the Quotation/SalesOrder/SalesInvoice which is to be deleted.

### PR DESCRIPTION
**Pull request by Mutantpenguin**
closes #37

please also backport to version-14

### CODE REVIEW

File: erpnext_germany/custom/sales.py [hunk 1]
```diff
@@ -16,11 +16,12 @@ def on_trash(doc: SellingController, event: str = None) -> None:
...
 
 
-def is_not_latest(doctype, name, creation):
+def is_not_latest(doctype, name, creation, company):
 	return frappe.db.exists(
 		doctype,
 		{
 			"creation": (">", creation),
 			"name": ("!=", name),
+			"company": ("=", company),
```
Comment (barredterra) [inline erpnext_germany/custom/sales.py]:
```suggestion
			"company": company,
```

The "=" is implicit

**Comment by barredterra** [comment]
General thought: shouldn't this be per naming series instead of per company? In theory, two companies could use the same naming series and then this code would permit gaps.

But I guess it's okay to assume that his won't happen.

**Comment by Mutantpenguin** [comment]
> General thought: shouldn't this be per naming series instead of per company? In theory, two companies could use the same naming series and then this code would permit gaps.
> 
> But I guess it's okay to assume that his won't happen.

I thought about this for a while too, but I can't think of a scenario where this would make sense.

I guess it could be easily added if someone really needs this.

**Review by barredterra — APPROVED** [review]

**Comment by barredterra** [comment]
@mergifyio backport version-14

**Comment by Mutantpenguin** [comment]
Thank you very much, that was fast :-)

**Pull request merged**